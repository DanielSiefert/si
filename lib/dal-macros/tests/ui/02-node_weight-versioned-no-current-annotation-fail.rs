pub struct InnerType;

pub trait SiVersionedNodeWeight {
    type Inner;

    fn inner(&self) -> &Self::Inner;
    fn inner_mut(&mut self) -> &mut Self::Inner;
}

#[derive(dal_macros::SiVersionedNodeWeight)]
pub enum VersionedNodeWeight {
    V0,
    V1(InnerType),
}

fn main() {}
