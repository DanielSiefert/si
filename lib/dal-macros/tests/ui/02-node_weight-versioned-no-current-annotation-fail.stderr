error: No current variant annotation
  --> tests/ui/02-node_weight-versioned-no-current-annotation-fail.rs:11:1
   |
11 | / pub enum VersionedNodeWeight {
12 | |     V0,
13 | |     V1(InnerType),
14 | | }
   | |_^
